content/kubernetes-platform-services/artifact-storage/chartmuseum.md

```diff
@@ -0,0 +1,122 @@
+
+Change directory to `/conf/kops-aws-platform` and run there commands to provision the `chart-repo` backend.
+```bash
+init-terraform
+terraform plan
+terraform apply
+```
+
+From the Terraform outputs, copy the following values to environment variables
+* `kops_chart_repo_bucket_bucket_id` -> `CHARTMUSEUM_STORAGE_AMAZON_BUCKET`
+* `kops_chart_repo_bucket_role_name` -> `CHARTMUSEUM_IAM_ROLE`
+
+{{% include-code-block title="terraform apply" file="kubernetes-platform-services/artifact-storage/examples/terraform-apply-chart-repo.txt" %}}
+
+In the example the bucket name is `kops_chart_repo_bucket_bucket_id = example-staging-chart-repo`.
+IAM role is `kops_chart_repo_bucket_role_name = example-staging-chart-repo`.
+
+## Install Chart
+
+To install the `chartmuseum`, you will need to define the `hostname`, which is the FQHN used to access the `chartmuseum`.
+
+In our example, we use `charts.us-west-2.staging.example.com` as the FQHN. Replace this with an appropriate value to suit your specific project.
+
+You can install `chartmuseum` in a few different ways, but we recommend using the [Master Helmfile](https://github.com/cloudposse/geodesic/blob/master/rootfs/conf/kops/helmfile.yaml).
+
+### Install with Master Helmfile
+
+Master Helmfile provide two releases of chartmuseum:
+* `charts` - Chartmuseum that serve charts
+* `charts-api` - Chartmuseum that provide api gateway to publish charts.
+
+These releases shares the same environment variables.
+`charts-api` gateway would be available on subdomain `api` for FQHN.
+In our example it would `api.charts.us-west-2.staging.example.com`.
+
+To install releases follow these instructions:
+1. Set the `CHARTMUSEUM_STORAGE_AMAZON_BUCKET` secret with chamber to copied from output value
+2. Set the `CHARTMUSEUM_STORAGE_AMAZON_REGION` secret with chamber
+3. Set the `CHARTMUSEUM_IAM_ROLE` secret with chamber to copied from output value
+4. Set the `CHARTMUSEUM_INGRESS` secret with chamber provided by [Nginx ingress]({{< relref "kubernetes-backing-services/ingress/nginx-ingress-controller.md" >}})
+5. Set the `CHARTMUSEUM_HOSTNAME` secret with chamber
+6. Run command `helmfile sync` to install `chartmuseum`.
+
+{{% dialog type="code-block" icon="fa fa-code" title="Install chartmuseum" %}}
+```
+chamber write kops CHARTMUSEUM_STORAGE_AMAZON_BUCKET example-staging-chart-repo
+chamber write kops CHARTMUSEUM_STORAGE_AMAZON_REGION us-west-2
+chamber write kops CHARTMUSEUM_IAM_ROLE example-staging-chart-repo
+chamber write kops CHARTMUSEUM_INGRESS ingress.us-west-2.staging.example.com
+chamber write kops CHARTMUSEUM_HOSTNAME charts.us-west-2.staging.example.com
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=chartmuseum sync
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=chartmuseum-api sync
+```
+{{% /dialog %}}
+
+These are some of the environment variables you may want to configure:
```

Comment:
These are the environment variables you may want to set to configure `chartmuseum`: